pkg/types/powervs/machinepools.go

```diff
@@ -20,21 +12,21 @@ type MachinePool struct {
 	// +optional
 	VolumeIDs []string `json:"volumeIDs,omitempty"`
 
-	// Memory defines the memory in GB for the instance.
+	// memoryGiB is the size of a virtual machine's memory, in GiB.
 	//
 	// +optional
-	Memory string `json:"memory,omitempty"`
+	MemoryGiB int32 `json:"memoryGiB,omitempty"`
 
 	// Processors defines the processing units for the instance.
 	//
 	// +optional
-	Processors string `json:"processors,omitempty"`
+	Processors intstr.IntOrString `json:"processors,omitempty"`
 
 	// ProcType defines the processor sharing model for the instance.
-	// Must be one of {capped, dedicated, shared}.
+	// Must be one of {Capped, Dedicated, Shared}.
 	//
 	// +optional
-	ProcType ProcType `json:"procType,omitempty"`
+	ProcType machinev1.PowerVSProcessorType `json:"procType,omitempty"`
```

Comment:
wondering if we need to add this here in that case:
```golang
// +kubebuilder:validation:Enum:="Dedicated";"Shared";"Capped";""
```

Reply:
Added it in the latest commit